### PR TITLE
Migrate from MDX + Docs Navigaton enhancements

### LINKED ISSUE #103: Remove MDX and restructure How the navigation works in the docs.

### PR DESCRIPTION
**Pull request by nisarhassan12**
Fixes #103 , #39 

The site now have syntax highlighting. 

image

### CODE REVIEW

File: src/components/Footer.js [hunk 1]
```diff
@@ -53,7 +53,7 @@ const Footer = ({background}) => (
                 </a>
             </div>
             <p className="footer__copyright"><a target="_blank" rel="noopener noreferrer" href=" http://www.eclipse.org/legal/privacy.php">Privacy Policy</a> | <a target="_blank" rel="noopener noreferrer" href=" http://www.eclipse.org/legal/termsofuse.php">Terms of Use</a> | <a target="_blank" rel="noopener noreferrer" href="http://www.eclipse.org/legal/copyright.php">Copyright Agent</a></p>
-            <p>© {(new Date()).getFullYear()} by <a href="https://typefox.io" target="_blank" rel="noopener">TypeFox</a></p>
+            <p>© {(new Date()).getFullYear()} by <a href="https://typefox.io" target="_blank" rel="noopener noreferrer">TypeFox</a></p>
```
Comment (jankeromnes) [inline src/components/Footer.js:56]: The canonical URL seems to be https://www.typefox.io/

Also, it's fine to have a referrer here.

File: src/components/Nav.js [hunk 2]
```diff
@@ -187,10 +186,10 @@ class Nav extends React.Component {
                             <a href="https://spectrum.chat/theia" target="_blank" rel="noopener noreferrer" className="nav__link">Community</a>
                         </li>
                         <li className="nav__item">
-                            <a href="https://www.typefox.io/theia/" className="nav__link" target="_blank" rel="noopener">Support</a>
+                            <a href="https://www.typefox.io/theia/" className="nav__link" target="_blank" rel="noreferrer noopener">Support</a>
```
Comment (jankeromnes) [inline src/components/Nav.js:189]: Having a referrer here is fine.

File: src/components/Nav.js [hunk 3]
```diff
@@ -187,10 +186,10 @@ class Nav extends React.Component {
...
                         </li>
                         <li className="nav__item">
-                            <a href="https://www.typefox.io/trainings/" className="nav__link" target="_blank" rel="noopener">Training</a>
+                            <a href="https://www.typefox.io/trainings/" className="nav__link" target="_blank" rel="noreferrer noopener">Training</a>
```
Comment (jankeromnes) [inline src/components/Nav.js:192]: Here too.

File: src/components/index/ContributorsAndAdopters.js [hunk 4]
```diff
@@ -78,7 +78,7 @@ const ContributorsAndAdopters = () => (
                     {
                         contributorsAndAdopters.map((item, i) => (
                             <div key={i} className="contributors__image-container">
-                                <a target="_blank" rel="noopener" href={item.href}>
+                                <a target="_blank" rel="noopener noreferrer" href={item.href}>
```
Comment (jankeromnes) [inline src/components/index/ContributorsAndAdopters.js:81]: 👍

File: src/components/index/Features.js [hunk 5]
```diff
@@ -53,9 +52,12 @@ const Features = () => (
     <StyledFeatures>
         <section className="features" id="features">
             <div className="row feature__container">
-                {
-                    features.map((feature, i) => <Feature {...feature} id={i}/>)
-                }
+                {features.map(
+                    (feature, i) => {
+                        console.log(i)
```
Comment (jankeromnes) [inline src/components/index/Features.js:57]: Is this really required in the Theia website?

File: src/components/index/Header.js [hunk 6]
```diff
@@ -72,11 +72,11 @@ const Header = () => (
                     <h2 className="heading-tertiary" style={{ fontSize: '2.2rem' }}>
                         Eclipse Theia is an extensible platform to develop multi-language Cloud & Desktop IDEs with state-of-the-art web technologies.
                         <br/>
-                        <a href="https://dev.to/svenefftinge/theia-1-0-finally-a-good-browser-ide-3ok0">Learn about the 1.0 Release!</a>
+                        <a href="https://dev.to/svenefftinge/theia-1-0-finally-a-good-browser-ide-3ok0" rel="noopener noreferrer">Learn about the 1.0 Release!</a>
                     </h2>
 
                     <a className="btn" href="https://github.com/eclipse-theia/theia" target="_blank" rel="noopener noreferrer">View on GitHub</a>
-                    <a className="btn btn--cta" href="https://gitpod.io/#https://github.com/eclipse-theia/theia" target="_blank" rel="noopener">Try in Gitpod &nbsp;&nbsp;&rarr;</a>
+                    <a className="btn btn--cta" href="https://gitpod.io/#https://github.com/eclipse-theia/theia" target="_blank" rel="noopener noreferrer">Try in Gitpod &nbsp;&nbsp;&rarr;</a>
```
Comment (jankeromnes) [inline src/components/index/Header.js:79]: It's fine to have a referrer here.

File: src/docs/composing_applications.md [hunk 7]
```diff
@@ -3,8 +3,6 @@ title: Build your own IDE
 slug: composing_applications
```
Comment (jankeromnes) [inline src/docs/composing_applications.md:3]: You seem to remove the `slug` everywhere else, but not here. Why not? 🙂

File: src/docs/json_rpc.md [hunk 8]
```diff
@@ -3,9 +3,6 @@ title: Communication via JSON-RPC
 slug: json_rpc
```
Comment (jankeromnes) [inline src/docs/json_rpc.md:3]: Maybe remove?

File: src/docs/language_support.md [hunk 9]
```diff
@@ -0,0 +1,10 @@
+---
+title: Adding Language Support
+slug: language_support
```
Comment (jankeromnes) [inline src/docs/language_support.md:3]: Why is the `slug` needed here?

File: src/components/Footer.js (hunk 1 above)
Comment (nisarhassan12) [reply src/components/Footer.js:56]: Thanks! I have removed the `noreferrer`. Sorry, I'm not able to understand the canonical is `https://theia-ide.org/`?  or do you want me to put a `/` at the end of the URL ?

File: src/docs/composing_applications.md (hunk 7 above)
Comment (nisarhassan12) [reply src/docs/composing_applications.md:3]: Thanks! somehow I forgot to remove this.

**Comment by nisarhassan12** [comment]
@jankeromnes I have made the changes. Please have a look again whenever you'll have time.

File: src/docs/menu.js [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+    ),
+    {
+        title: 'Using Theia'
+    },
+    M(
+        'Build your own IDE',
+        'composing_applications'
+    ),
+    M(
+        'Authoring Theia Extensions',
```
Comment (jankeromnes) [inline src/docs/menu.js:32]: Nit: I found the previous title a bit nicer (shorter):

```suggestion
        'Authoring an Extension',
```

File: src/docs/menu.js [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+        'authoring_extensions'
+    ),
+    M(
+        'Authoring Theia Plug-ins',
```
Comment (jankeromnes) [inline src/docs/menu.js:36]:
```suggestion
        'Authoring Plug-ins',
```

**Pull request merged**